flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -68,6 +72,28 @@
 public class TaskManagerServices {
     private static final Logger LOG = LoggerFactory.getLogger(TaskManagerServices.class);
 
+    /**
+     * This is an expert option, that we do not want to expose in the documentation. The default
+     * value is good enough for almost all cases
+     */
+    @Experimental
+    private static final ConfigOption<Duration> SHUFFLE_DESCRIPTORS_CACHE_EXPIRE_TIMEOUT =
+            ConfigOptions.key("taskmanager.shuffle-descriptors-cache.expire-timeout")
+                    .durationType()
+                    .defaultValue(Duration.ofSeconds(300))
+                    .withDescription("The expire timeout of shuffle descriptor caches.");
+
+    /**
+     * This is an expert option, that we do not want to expose in the documentation. The default
+     * value is good enough for almost all cases
+     */
+    @Experimental
+    private static final ConfigOption<Integer> SHUFFLE_DESCRIPTORS_CACHE_SIZE_LIMIT =
+            ConfigOptions.key("taskmanager.shuffle-descriptors-cache.size-limit")
+                    .intType()
+                    .defaultValue(10)
+                    .withDescription("The size limit of shuffle descriptor caches.");
+
```

Comment:
I tend not to introduce configuration options for these two parameters. We can provide a new ctor `DefaultShuffleDescriptorsCache()`, and using the default value "300" and "10" in that ctor.